lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -38,100 +38,44 @@
-    associated with the master coordinate.
-
-    For example, forecast_reference_time and forecast_period can be converted
-    from scalar coordinates to auxiliary coordinates, and associated with time.
+    Function to remove attributes that do not match between all cubes on the
+    list.  Cubes are modified in place.
 
     Args:
-        cube (Iris cube):
-            Cube requiring addition of the specified coordinates as auxiliary
-            coordinates.
-        master_coord (String):
-            Coordinate that the other coordinates will be associated with.
-        coordinates (None or List):
-            List of coordinates to be associated with the master_coord.
-
-    Returns:
-        forecast_data (Iris cube):
-            Cube where the the requested coordinates have been added to the
-            cube as auxiliary coordinates and associated with the desired
-            master coordinate.
-
-    Raises:
-        ValueError: If the master coordinate is not present on the cube.
-
-    """
-    if coordinates is None:
-        coordinates = []
-
-    if not cube.coords(master_coord):
-        msg = (
-            "The master coordinate for associating other "
-            "coordinates with is not present: "
-            "master_coord: {}, other coordinates: {}".format(
-                master_coord, coordinates))
-        raise ValueError(msg)
-
-    # If the master_coord is not a dimension coordinate, then the other
-    # coordinates cannot be associated with it.
-    if len(cube.coords(master_coord, dim_coords=True)) > 0:
-        for coord in coordinates:
-            if cube.coords(coord):
-                temp_coord = cube.coord(coord)
-                cube.remove_coord(coord)
-                temp_aux_coord = (
-                    build_coordinate(temp_coord.points,
-                                     bounds=temp_coord.bounds,
-                                     coord_type=AuxCoord,
-                                     template_coord=temp_coord))
-                coord_names = [
-                    coord.standard_name for coord in cube.dim_coords]
-                cube.add_aux_coord(
-                    temp_aux_coord,
-                    data_dims=coord_names.index(master_coord))
-    return cube
-
-
-def _slice_over_coordinate(cubes, coord_to_slice_over):
-    """
-    Function slice over the requested coordinate,
-    promote the sliced coordinate into a dimension coordinate
-    to help concatenation.
-
-    Args:
-        cubes (Iris cubelist or Iris cube):
-            Cubes to be concatenated.
-        coords_to_slice_over (List):
-            Coordinates to be sliced over.
+        cubes (iris.cube.CubeList):
+            List of cubes to check the attributes and revise.
 
-    Returns:
-        Iris CubeList
-            CubeList containing sliced cubes.
+    Kwargs:
+        unmatched (list or None):
+            List of unmatched attributes.  If None, function
```

Comment:
Based on this description, I would have expected to pass something like `unmatched = ["history"]` into the function, which would have resulted in the `history` attribute being removed. However, based on the unit tests, it seems like this function is actually expecting something as shown below where unmatched attributes is a list containing a dictionary, which matches the output from the `compare_attributes` function:
```python
        unmatched_attributes = [
            {"history": "2017-01-18T08:59:53: StaGE Decoupler"},
            {"history": "2017-01-18T08:59:53: StaGE Decoupler"}]
```
If the intention is to keep this keyword argument, then adding an example of what is expected would be useful.

Reply:
Good point.  Added to docstring.

Reply:
As noted above, I'm going to remove the `unmatched` option as it's new functionality that's not actually required.